### ISSUE TITLE
Typography: Improve type scaling

### ISSUE THREAD

**Issue by ItsJonQ** [ItsJonQ/g2#24]
At the moment, text size is being determined by the systems base font size (ultimately a CSS variable). In most cases, text is rendered with the Text component.

The Text component accepts arbitrary sizes for custom font size rendering, which results in px values.

Adaptive font size support, or rather lack of, is noticeable on mobile devices. It may also be problematic in scenarios where preferred font size is increased or decreases based on user preference (device based, browser based, config based)

With all that being said... the style system should (ultimately) output adaptive font sizes. However, the implementation detail of this should be invisible in the component layer.

For example, let's pretend the system base is 13px.

If we pass the Text component a font size of 14px.

The final computed font size would result in 14px. How it is derived, would be more like...

14 / 13, giving us an adaptive multiplier, which could be multiplied by a base value (eg 13)

**Comment by ItsJonQ** [comment]
Resolved with this!

https://github.com/ItsJonQ/g2/commit/0c589cb89e7c5d502754615fbdbae49715555891#diff-9260a6283c8d858b277ff9df3f6df154
